packages/v2/gui/src/shared/datafetcher/DataFetcher.tsx

```diff
@@ -0,0 +1,28 @@
+import { addLegacySerializerOption } from '@k9-sak-web/gui/utils/axios/axiosUtils.js';
+import { useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import React from 'react';
+
+interface DataFetcherProps {
+  url: string;
+  contentRenderer: (data: any, isLoading: boolean, hasError: boolean) => React.ReactNode;
+}
```

Comment:
```suggestion
interface DataFetcherProps<T> {
  url: string;
  contentRenderer: (data: T | undefined, isLoading: boolean, hasError: boolean) => React.ReactNode;
}
```
Kan vurdere å legge til en generisk type istedenfor `any` for bedre typesikkerhet.